content/blog/2022-07-26-schema-evolution-glue-msk/index.md

```diff
@@ -0,0 +1,861 @@
+                            </goals>
+                            <configuration>
+                                <sourceDirectory>${project.basedir}/src/main/resources/avro/</sourceDirectory>
+                                <outputDirectory>${project.basedir}/src/main/java/</outputDirectory>
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
+```
+
+While the root pom is a bit lengthy, the `pom.xml` files of the two modules are quite simple.
+They only reference the root pom, activate the `avro-maven-plugin`, and define a main class (we'll go into detail on the actual Java code in [the section below](#implement-a-producer-and-consumer)).
+
+The producer's `pom.xml`:
+{{< github repo="localstack/localstack-pro-samples" file="glue-msk-schema-registry/producer/pom.xml" lang="xml" ref="cd023a84a3b473984e9c34053d4feb7de8e038c1" >}}
+
+And analogous the consumer's `pom.xml`:
+{{< github repo="localstack/localstack-pro-samples" file="glue-msk-schema-registry/consumer/pom.xml" lang="xml" ref="cd023a84a3b473984e9c34053d4feb7de8e038c1" >}}
+
+Now the project is all set up and we can already generate our schema classes from the AVRO schema using the `avro-maven-plugin`:
+```
+mvn clean generate-sources
+```
+
+After the maven plugin is done, we have all types generated for both, the producer and the consumer:
+```plaintext
+.
+├── consumer
+│   └── src
+│       └── main
+│           ├── java
+│               └── cloud
+│                   └── localstack
+│                       └── demos
+│                           └── gluemsk
+│                               └── schema
+│                                   ├── Customer.java
+│                                   ├── ModeOfPayment.java
+│                                   ├── RecommendedUnicorn.java
+│                                   ├── unicorn_color.java
+│                                   ├── UnicornPreferredColor.java
+│                                   └── UnicornRideRequest.java
+└── producer
+    └── src
+        └── main
+            └── java
+                └── cloud
+                    └── localstack
+                        └── demos
+                            └── gluemsk
+                                └── schema
+                                    ├── Customer.java
+                                    ├── ModeOfPayment.java
+                                    ├── RecommendedUnicorn.java
+                                    ├── unicorn_color.java
+                                    ├── UnicornPreferredColor.java
+                                    └── UnicornRideRequest.java
+``` 
+
+Since we are just going log the sent and received messages, we need to configure the logging by adding a `log4j.properties` to the `src/main/resources` folder of the two modules:
```

Comment:
```suggestion
Since we want to log the sent and received messages, we need to configure the logging by adding a `log4j.properties` to the `src/main/resources` folder of the two modules:
```